### PR TITLE
DDF-5573 Upgraded dumbster to 1.7.1

### PR DESCRIPTION
**Pull request by shaundmorris**
#### What does this PR do?
Upgrades dumbster (SMTP client) to a new version. The previous version may have been the cause of occasional builds hanging in CI. 

#### Who is reviewing it? 
@glenhein 
@brjeter 

#### Select relevant component teams: 
@codice/test 

#### Ask 2 committers to review/merge the PR and tag them here.
@AzGoalie
@jrnorth 

#### How should this be tested?
Build platform-email-impl

#### What are the relevant tickets?
Fixes: #5573 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

---
This change is [Reviewable](https://reviewable.io/reviews/codice/ddf/5580)

### CODE REVIEW

File: platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java [hunk 1]
```diff
@@ -111,9 +114,10 @@ public void testSendWithAttachments()
 
     server.stop();
 
-    assertThat(server.getReceivedEmailSize(), is(1));
```
Comment (shaundmorris) [inline platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java]: Open to doing some of this better. I was just looking to do the minimum needed to upgrade the test depenendency

File: platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java [hunk 2]
```diff
@@ -111,9 +114,10 @@ public void testSendWithAttachments()
...
-    Iterator emailIterator = server.getReceivedEmail();
-    SmtpMessage email = (SmtpMessage) emailIterator.next();
+    List<SmtpMessage> emails = server.getReceivedEmails();
+    assertThat(emails, is(not(empty())));
+    SmtpMessage email = emails.get(0);
+    assertNotNull(email);
```
Comment (jrnorth) [inline platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java]: `assertNotNull("The email was null.", email)`

File: platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java [hunk 3]
```diff
@@ -154,9 +158,10 @@ public void testSend()
 
     server.stop();
 
-    assertThat(server.getReceivedEmailSize(), is(1));
-    Iterator emailIterator = server.getReceivedEmail();
-    SmtpMessage email = (SmtpMessage) emailIterator.next();
+    List<SmtpMessage> emails = server.getReceivedEmails();
+    assertThat(emails, is(not(empty())));
```
Comment (jrnorth) [inline platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java]: Same comments here.

File: platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java [hunk 4]
```diff
@@ -111,9 +114,10 @@ public void testSendWithAttachments()
...
```
[the remark this answers is not in the document]
Comment (shaundmorris) [reply platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java]: fixed

File: platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java (hunk 2 above)
Comment (shaundmorris) [reply platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java]: fixed

File: platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java (hunk 3 above)
Comment (shaundmorris) [reply platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/SmtpClientImplITCaseTest.java]: fixed

**Comment by shaundmorris** [comment]
build now

**Comment by shaundmorris** [comment]
full build timed out. Verified this was not caused by this module.

build now

**Comment by jrnorth** [comment]
build now

**Comment by shaundmorris** [comment]
build now

**Comment by shaundmorris** [comment]
build now

**Comment by shaundmorris** [comment]
build now

**Comment by shaundmorris** [comment]
build now

**Comment by shaundmorris** [comment]
manually aborted to try skipping some modules

build now

**Pull request merged**